Sources/Service/WalletAvailabilityService.swift

```diff
@@ -35,7 +47,11 @@ class WalletAvailabilityService: WalletFeatureAvailabilityService {
         return true
     }
     
-    func hasAccessedPreviously() {
-        UserDefaults.standard.set(true, forKey: "hasAccessedWalletBefore")
+    func accessedFeature() {
+        hasAccessedBefore = true
+    }
+    
+    func resetFeatureState() {
+        hasAccessedBefore = false
     }
```

Comment:
So, even though `hasAccessedBefore` is a var, when trying to set a new value by accessing the instance of walletAvailibilityService - it would not allow, and it would require that instance to be changed to a var. Not sure if I explained it right, but I can show this if needed

Reply:
That should work if you conform FeatureAvailabilityService to `AnyObject` !
`protocol FeatureAvailabilityService: AnyObject {`

Reply:
Fab 🙂 presumably the `accessedFeature` and `resetFeatureState` aren't required anymore, we can just update the boolean value?